### PR TITLE
Add basic update event

### PR DESCRIPTION
**Pull request by efratush**
#35

### CODE REVIEW

File: app/routers/event.py [hunk 1]
```diff
@@ -19,3 +24,73 @@ async def eventedit(request: Request):
...
+def get_event_by_id(db: Session, event_id: int) -> Event:
+    """Select event by id"""
+    return db.query(Event).filter(Event.id == event_id).first()
+
+
+# copy from michael ben david
+def check_validation(start_time: datetime, end_time: datetime) -> bool:
+    """Check if the start time is earlier than the end time"""
+
+    if start_time < end_time:
```
Comment (yammesicka) [inline app/routers/event.py:38]: Prefer: `return start_time < end_time`.

File: app/routers/event.py [hunk 2]
```diff
@@ -19,3 +24,73 @@ async def eventedit(request: Request):
...
                                       {"request": request, "event_id": id})
+
+
+def get_event_by_id(db: Session, event_id: int) -> Event:
+    """Select event by id"""
+    return db.query(Event).filter(Event.id == event_id).first()
+
+
+# copy from michael ben david
+def check_validation(start_time: datetime, end_time: datetime) -> bool:
```
Comment (yammesicka) [inline app/routers/event.py:35]: Prefer a informative naming: `is_later_date`

File: app/routers/event.py [hunk 3]
```diff
@@ -19,3 +24,73 @@ async def eventedit(request: Request):
...
+    """Check if the start time is earlier than the end time"""
+
+    if start_time < end_time:
+        return True
+    return False
+
+
+def get_column_names_event(db: Session) -> Set:
+    result = db.execute("select * from events")
+    return {col for col in result.keys()}
```
Comment (yammesicka) [inline app/routers/event.py:45]: Prefer using `set()`

(Great job choosing `set`. btw)

File: app/routers/event.py [hunk 4]
```diff
@@ -19,3 +24,73 @@ async def eventedit(request: Request):
...
+def check_validation(start_time: datetime, end_time: datetime) -> bool:
+    """Check if the start time is earlier than the end time"""
+
+    if start_time < end_time:
+        return True
+    return False
+
+
+def get_column_names_event(db: Session) -> Set:
+    result = db.execute("select * from events")
```
Comment (yammesicka) [inline app/routers/event.py:44]: Prefer using the ORM instead.

Also, If possible, try to get the column names without selecting the whole table, which might be a very heavy query.

File: app/routers/event.py [hunk 5]
```diff
@@ -19,3 +24,73 @@ async def eventedit(request: Request):
...
+                if not check_validation(event_update.start, old_event.end):
+                    return False
+            elif not check_validation(old_event.start, event_update.end):
+                return False
+        except TypeError:
+            return False
+    return True
+
+
+def update_event(event_id: int, event_items: Dict, db: Session
```
Comment (yammesicka) [inline app/routers/event.py:65]: This function is really long. Can we somehow split it into multiple functions?

File: app/routers/event.py [hunk 6]
```diff
@@ -19,3 +24,73 @@ async def eventedit(request: Request):
...
+    try:
+        old_event = get_event_by_id(db=db, event_id=event_id)
+    except AttributeError:  # Problem connecting to db
+        return None
+
+    if old_event is None:   # No such event number.
+        return None
+
+    column_names = get_column_names_event(db)
+    for key in event_items.keys():
```
Comment (yammesicka) [inline app/routers/event.py:82]: You can leave out the `.keys`

File: app/routers/event.py [hunk 7]
```diff
@@ -19,3 +24,73 @@ async def eventedit(request: Request):
...
+        if key not in column_names:
```
Comment (yammesicka) [inline app/routers/event.py:83]: Another nice, readable trick might be:

```
if any(key not in column_names for key in event_items):
    return None
```

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: tests/test_event.py [hunk 8]
```diff
@@ -1,22 +1,58 @@
...
+def test_eventedit(client):
     response = client.get("/event/edit")
     assert response.status_code == 200
     assert b"Edit Event" in response.content
 
 
-def test_eventview_with_id():
+def test_eventview_with_id(client):
     response = client.get("/event/view/1")
     assert response.status_code == 200
```
Comment (ivarshav) [inline tests/test_event.py:23]: Please use starlette.status for status code

File: tests/test_event.py [hunk 9]
```diff
@@ -1,22 +1,58 @@
...
 
-def test_eventview_without_id():
+def test_eventview_without_id(client):
     response = client.get("/event/view")
     assert response.status_code == 404
+
+
+@pytest.mark.parametrize("data", INVALID_UPDATE_OPTIONS)
+def test_invalid_update(data, event, session):
+    assert update_event(1, data, session) is None
```
Comment (ivarshav) [inline tests/test_event.py:34]: What is 1? the event id?

Comment (efratush) [reply tests/test_event.py]: fixed

File: tests/test_event.py [hunk 10]
```diff
@@ -17,5 +28,36 @@ def test_eventview_without_id(self, client):
...
+                            event_dict=data, db=session) is None
+
+    @staticmethod
+    def test_successful_update(event, session):
+        data = {
+            "title": "successful",
+            "start": datetime(2021, 1, 20),
+            "end": datetime(2021, 1, 21),
+        }
+        assert type(update_event(1, data, session)) == Event
```
Comment (yammesicka) [inline tests/test_event.py:44]: Use `isinstance` instead :)

File: app/routers/event.py [hunk 11]
```diff
@@ -27,6 +28,41 @@ async def eventview(request: Request, id: int):
...
+    """Select event by id"""
+    return db.query(Event).filter(Event.id == event_id).first()
+
+
+def validate_dates(start_date: datetime, end_date: datetime) -> bool:
+    """Check if the start date is earlier than the end date"""
+    return start_date < end_date
+
+
+def update_event(event_id: int, event_dict: Dict, db: Session
```
Comment (yammesicka) [inline app/routers/event.py:41]: Maybe just `events`? We usually try not to use [Hungarian Notation](https://en.wikipedia.org/wiki/Hungarian_notation) when it's possible :)

File: app/routers/event.py [hunk 12]
```diff
@@ -27,6 +28,41 @@ async def eventview(request: Request, id: int):
                                       {"request": request, "event_id": id})
 
 
+def get_event_by_id(db: Session, event_id: int) -> Event:
```
Comment (yammesicka) [inline app/routers/event.py:31]: I'd event call it `by_id` to allow external modules use `event.by_id`

**Review by yammesicka — COMMENTED** [review]
Great job! Added few of my insights :)

File: app/routers/event.py (hunk 11 above)
Comment (efratush) [reply app/routers/event.py:41]: *event

File: app/routers/event.py [hunk 13]
```diff
@@ -27,6 +28,41 @@ async def eventview(request: Request, id: int):
                                       {"request": request, "event_id": id})
 
 
+def by_id(db: Session, event_id: int) -> Event:
+    """Select event by id"""
+    return db.query(Event).filter(Event.id == event_id).first()
+
+
+def validate_dates(start_date: datetime, end_date: datetime) -> bool:
```
Comment (yammesicka) [inline app/routers/event.py:36]: Maybe `is_date_before`?

File: app/routers/event.py [hunk 14]
```diff
@@ -27,6 +28,41 @@ async def eventview(request: Request, id: int):
...
+    """Check if the start date is earlier than the end date"""
+    return start_date < end_date
+
+
+def update_event(event_id: int, event: Dict, db: Session
```
Comment (yammesicka) [inline app/routers/event.py:41]: This function does more than one thing. Can you please separate it to 2-3 function?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/routers/event.py [hunk 15]
```diff
@@ -27,6 +28,41 @@ async def eventview(request: Request, id: int):
...
+                 ) -> Optional[Event]:
+    # TODO Check if the user is the owner of the event.
+
+    # Extract only that keys to update
+    event_to_update = {i: event[i] for i in (
+        'title', 'start', 'end', 'content', 'location') if i in event}
+    if not bool(event_to_update):  # Event items is empty
```
Comment (yammesicka) [inline app/routers/event.py:48]: Is the `bool` here necessary?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/routers/event.py (hunk 15 above)
Comment (efratush) [reply app/routers/event.py:48]: You're right.

**Review by yammesicka — APPROVED** [review]

**Pull request merged**